### PR TITLE
add cni plugins to hyperkube image

### LINKED ISSUE #24874: hyperkube should package cni plugins
cc @freehan

### PR DESCRIPTION
**Pull request by mikedanese**
fixes #24874

### CODE REVIEW

**Comment by tomdee** [comment]
What cni release is this using? The new one from today?

**Comment by tomdee** [comment]
Also, the directory is wrong. The files need to be downloaded to /opt/cni/bin https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni.go#L36

File: cluster/images/hyperkube/Dockerfile [hunk 1]
```diff
@@ -53,6 +53,10 @@ COPY make-ca-cert.sh /make-ca-cert.sh
 # easy-rsa package required by make-ca-cert
 ADD https://storage.googleapis.com/kubernetes-release/easy-rsa/easy-rsa.tar.gz /root/kube/
 
+RUN mkdir -p /opt/cni
+RUN curl https://storage.googleapis.com/kubernetes-release/network-plugins/cni-42c4cb842dad606a84e93aad5a4484ded48e3046.tar.gz \
+  | tar xzfv - -C /opt/cni
```
Comment (tomdee) [inline cluster/images/hyperkube/Dockerfile]: If you drop the f flag, you don't need the `-` to read from stdin

**Comment by freehan** [comment]
@tomdee The CNI binaries on `kubernetes-release` is different than the ones from the cni repo. Last time I check, theirs were dynamically linked. CNI is also in the process of improving their release process so that we could use it with our built. 

The package from `kubernetes-release` already contains the `bin` folder

**Comment by mikedanese** [comment]
Yup, looks built from https://github.com/appc/cni/tree/42c4cb842dad606a84e93aad5a4484ded48e3046

**Comment by tomdee** [comment]
Cool, that makes sense.

**Comment by mikedanese** [comment]
As it turns out, this actually doesn't work very well.... Would be happy for some debugging help.

**Comment by cheld** [comment]
CC @zreigz, @batikanu

**Comment by mikedanese** [comment]
@k8s-bot test this issue: #hashtag

**Comment by mikedanese** [comment]
failures caused by build breakage yesterday

**Comment by freehan** [comment]
I just bump the cni version. https://github.com/kubernetes/kubernetes/pull/25175

It solves a bunch of problems.

**Comment by freehan** [comment]
Looks like k8s-merge-robot is replying in a loop. LOL

**Comment by mikedanese** [comment]
@eparis k8s-merge-robot is broken

**Comment by mikedanese** [comment]
@freehan updated

**Comment by freehan** [comment]
LGTM

**Comment by luxas** [comment]
Will this make a docker restart unneccesary when e.g. deploying docker-multinode w/ flannel?

**Comment by mikedanese** [comment]
Early breakage was fixed by https://github.com/kubernetes/kubernetes/issues/25092

**Comment by mikedanese** [comment]
@luxas that's the dream!

**Pull request merged**